### PR TITLE
ref: Split code into more modules

### PR DESCRIPTION
**Pull request by TBS1996**
also created the WaitingUsers abstraction to make this easier.

### CODE REVIEW

File: src/frontend/mod.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    }
+    if !(0. ..=100.).contains(&a) {
+        return None;
+    }
+    if !(0. ..=100.).contains(&n) {
+        return None;
+    }
+
+    Some(Scores { o, c, e, a, n })
+}
```
Comment (henriquekirchheck) [inline src/frontend/mod.rs:79]: Can't this be done by implementing `impl TryFrom<FormData> for Score`?

**Review by henriquekirchheck — CHANGES_REQUESTED** [review]
Making it more idiomatic for Rust should make it more concise

File: src/frontend/mod.rs (hunk 1 above)
Comment (TBS1996) [reply src/frontend/mod.rs:79]: sure, but maybe in another PR ? this one is kinda big already so let's not introduce more complexities than necessary

Comment (henriquekirchheck) [reply src/frontend/mod.rs:79]: Alright, I'll do it in another pr

**Review by henriquekirchheck — APPROVED** [review]

**Pull request merged**